expression/builtin_cast.go

```diff
@@ -977,6 +977,10 @@ func (b *builtinCastDecimalAsTimeSig) evalTime(row chunk.Row) (res types.Time, i
 	if isNull || err != nil {
 		return res, isNull, err
 	}
+	// MySQL compatibility: 0.0 should not be converted to null, see #11203
+	if val.IsZero() {
+		return types.Time{}, false, nil
+	}
```

Comment:
Could this judgement also be put into `types.ParseTimeFromFloatString`?

Reply:
Done